motionplan/motionPlanner.go

```diff
@@ -49,6 +49,7 @@ type PlanRequest struct {
 	StartPose          spatialmath.Pose
 	StartConfiguration map[string][]frame.Input
 	WorldState         *frame.WorldState
+	BoundingRegions    []spatialmath.Geometry
```

Comment:
This feels like something that could go into WorldState maybe; @raybjork thoughts?

Reply:
hmmmmm, this is an important thing to consider - good call out.

I am slightly leaning towards BoundingRegions remaining as a separate entity from the WorldState.
My reasoning is that if we were to add bounding regions into the worldstate I feel that this would undo work done by the "Remove Interaction Spaces from Worldstate" project completed last year.

If we were to include it in the worldstate we then are tasked with another question: Do we update the protomessage of worldstate?

Also curious to hear your thoughts on this, I am open either way :)

Reply:
After some deliberation I am still leaning towards having bounding regions remain separate from the worldstate.

My reasoning is that a user may specify a bounding region for a worldstate they pass into a motionservice.Move call. Since our worldstate proto message does not contain an bounding regions field, this information would not be correctly passed along. And so the expected behavior would be violated.

Reply:
https://viam.atlassian.net/browse/RSDK-7802

Reply:
That sounds good to me

Reply:
I agree we should continue to mirror the struct and proto; the bounding regions could be added to that worldState as well.